### PR TITLE
Make compatible with rubocop 1.68.0

### PR DESCRIPTION
**Pull request by dduugg**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----
See https://github.com/Homebrew/brew/pull/18682

### CODE REVIEW

File: Formula/p/portaudio.rb [hunk 1]
```diff
@@ -13,7 +13,7 @@ class Portaudio < Formula
     regex(/href=.*?pa[._-]stable[._-]v?(\d+)(?:[._-]\d+)?\.t/i)
     strategy :page_match do |page, regex|
       # Modify filename version (190700) to match formula version (19.7.0)
-      page.scan(regex).map { |match| match&.first&.scan(/\d{2}/)&.map(&:to_i)&.join(".") }
+      page.scan(regex).map { |match| match.first.scan(/\d{2}/).map(&:to_i).join(".") }
```
Comment (dduugg) [inline Formula/p/portaudio.rb:16]: The nil-forwarding here should be unnecessary cc @samford (tested with current page)

Comment (samford) [reply Formula/p/portaudio.rb:16]:
```suggestion
      page.scan(regex).map { |match| match[0].scan(/\d{2}/).map(&:to_i).join(".") }
```

Yep, `match` will always be present in this context when the regex matches. Similarly, the capture group isn't optional, so `match[0]` will also be present when the regex matches.

I recently cleaned up similar instances in homebrew/cask, removing unnecessary safe navigation operators and using numeric addressing (i.e., `match.first` -> `match[0]`). I'll create a follow-up PR to address the same for homebrew/core in a bit.

Comment (dduugg) [reply Formula/p/portaudio.rb:16]: Works for me (it's a better (typesafe) to use `.fetch` if non-nil is guaranteed, though that can be a bit pedantic)

**Review by samford — APPROVED** [review]

File: Formula/g/gbdfed.rb [hunk 2]
```diff
@@ -57,4 +57,4 @@ def install
 +	-DGDK_PIXBUF_DISABLE_DEPRECATED
  
  SRCS = bdf.c \
-        bdfcons.c \
+        bdfcons.c
```
Comment (Bo98) [inline Formula/g/gbdfed.rb:60]: This is not Ruby code and sounds like a bug in RuboCop

File: Formula/l/lua@5.3.rb [hunk 3]
```diff
@@ -187,4 +187,4 @@ def caveats
 +	$(MAKE) $(ALL) SYSCFLAGS="-DLUA_USE_MACOSX -fno-common" SYSLIBS="-lreadline"
 
  mingw:
-	$(MAKE) "LUA_A=lua53.dll" "LUA_T=lua.exe" \
+	$(MAKE) "LUA_A=lua53.dll" "LUA_T=lua.exe"
```
Comment (Bo98) [inline Formula/l/lua@5.3.rb:190]: ditto

File: Formula/r/regldg.rb [hunk 4]
```diff
@@ -49,4 +49,4 @@ def install
 +COMPILE=cc -O3 -Wall -Wno-int-conversion -g -c
  LINK=gcc -O3 -Wall -g -lm
 
- all: alt.o altlist.o build_structs.o char_set.o data.o debug.o \
+ all: alt.o altlist.o build_structs.o char_set.o data.o debug.o
```
Comment (Bo98) [inline Formula/r/regldg.rb:52]: ditto

**Review by Bo98 — CHANGES_REQUESTED** [review]

File: Formula/g/gbdfed.rb (hunk 2 above)
Comment (Bo98) [reply Formula/g/gbdfed.rb:60]: Thanks! Happy to merge the `portaudio` and `util-linux` changes in the meantime and we can wait for 1.68.1 or something.

**Review by Bo98 — APPROVED** [review]

**Review by carlocab — APPROVED** [review]

**Pull request merged**